File: platform/src/main/java/com/nixvision/support/terminal/platform/app/domains/bot/commands/domains/bot/command/requests/BotCommandRequest.java
package com.nixvision.support.terminal.platform.app.domains.bot.commands.domains.bot.command.requests;



import com.nixvision.support.terminal.platform.app.domains.core.action.models.actions.Action;
import lombok.Data;

import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.Collection;

@Data
public class BotCommandRequest {

    @NotBlank
    private String name;

    @NotBlank
    private String cmd;

    @NotNull
    @Valid
    private Action action;

    @NotEmpty
    private Collection<String> botIds;

    private String state;

}

File: src/main/kotlin/io/github/support/terminal/application/domains/bot_commands/rest/BotCommandRequest.kt
package io.github.support.terminal.application.domains.bot_commands.rest

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import io.github.support.terminal.application.domains.common.action.ActionRequest
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

@JsonIgnoreProperties(ignoreUnknown = true)
data class BotCommandRequest(
        val name: @NotBlank String,
        val cmdTemplate: @NotBlank String,
        val action: @NotNull ActionRequest,
        val botIds: @NotEmpty Collection<String> = ArrayList(),
        val enabled: Boolean
)